### PR TITLE
sql: test ON UPDATE's interaction with backfills over row

### LINKED ISSUE #69058: sql: test ON UPDATE interactions
Upon completion of #69057, we will need to test ON UPDATE interactions with other types of constraints. Of interest are:
- Interaction with foreign key updates
- Interaction with DEFAULT expressions
- Interaction with backfills

Epic CRDB-9148

### PR DESCRIPTION
**Pull request by ZhouXing19**
Fixes #69058

Release Note: none

### CODE REVIEW

File: pkg/sql/logictest/testdata/logic_test/on_update [hunk 1]
```diff
@@ -405,3 +405,48 @@ ALTER TYPE test_enum DROP VALUE 'x'
 
 statement error pq: could not remove enum value "y" as it is being used in an ON UPDATE expression of "test_array_enum"
 ALTER TYPE test_enum DROP VALUE 'y'
+
+# Interaction with backfills over row
+subtest SimpleUsage
```
Comment (otan) [inline pkg/sql/logictest/testdata/logic_test/on_update:410]: maybe `subtest test_on_update_backfill`

File: pkg/sql/logictest/testdata/logic_test/on_update [hunk 2]
```diff
@@ -405,3 +405,48 @@ ALTER TYPE test_enum DROP VALUE 'x'
 
 statement error pq: could not remove enum value "y" as it is being used in an ON UPDATE expression of "test_array_enum"
 ALTER TYPE test_enum DROP VALUE 'y'
+
+# Interaction with backfills over row
```
Comment (otan) [inline pkg/sql/logictest/testdata/logic_test/on_update:409]: let's make the comment "Tests adding new columns will not trigger the ON UPDATE clause."

Comment (ZhouXing19) [reply pkg/sql/logictest/testdata/logic_test/on_update]: Thanks! Done.

**Review by otan — APPROVED** [review]

**Comment by ZhouXing19** [comment]
Thanks!
bors r=otan

**Pull request merged**